benchmark/storage/varlen_entry_hash_benchmark.cpp

```diff
@@ -0,0 +1,101 @@
+#include <memory>
+#include <vector>
+
+#include "benchmark/benchmark.h"
+#include "storage/storage_defs.h"
+#include "test_util/storage_test_util.h"
+
+namespace terrier {
+
+class VarlenEntryHashBenchmark : public benchmark::Fixture {
+ public:
+  void SetUp(const benchmark::State &state) final {}
+
+  void TearDown(const benchmark::State &state) final {}
+
+  std::default_random_engine generator_;
+};
+
+#define VARLEN_HASH_BENCHMARK(varlen_bytes)                                                       \
+                                                                                                  \
+  byte random_buffer[(varlen_bytes)];                                                             \
+  StorageTestUtil::FillWithRandomBytes((varlen_bytes), random_buffer, &generator_);               \
+                                                                                                  \
+  if constexpr ((varlen_bytes) <= storage::VarlenEntry::InlineThreshold()) {                      \
+    const auto varlen_entry = storage::VarlenEntry::CreateInline(random_buffer, (varlen_bytes));  \
+    /* NOLINTNEXTLINE */                                                                          \
+    for (auto _ : state) {                                                                        \
+      benchmark::DoNotOptimize(varlen_entry.Hash());                                              \
+    }                                                                                             \
+  } else {                                                                                        \
+    const auto varlen_entry = storage::VarlenEntry::Create(random_buffer, (varlen_bytes), false); \
+    /* NOLINTNEXTLINE */                                                                          \
+    for (auto _ : state) {                                                                        \
+      benchmark::DoNotOptimize(varlen_entry.Hash());                                              \
+    }                                                                                             \
+  }                                                                                               \
+                                                                                                  \
+  state.SetItemsProcessed(state.iterations());                                                    \
+  state.SetBytesProcessed(state.iterations() * (varlen_bytes));
+
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(VarlenEntryHashBenchmark, Inline4Bytes)(benchmark::State &state) { VARLEN_HASH_BENCHMARK(4); }
```

Comment:
Instead of inline and non-inline versions for different lengths, why not just have two:`BENCHMARK_DEFINE_F(VarlenEntryHashBenchmark, Inline)` and `BENCHMARK_DEFINE_F(VarlenEntryHashBenchmark, NonInline)`. Then configure the benchmarks with length ranges, 1-12 for inlined, and 12+ otherwise? This will cut the number of benchmarks and can rid of the macro.

Reply:
How would that data get presented? We see very different performance at different VarlenEntry sizes, so it made more sense to test the sizes in separate benchmarks.